Task-5-database-in-go/controller/routeHandler.go

```diff
@@ -0,0 +1,76 @@
+package controller
+
+import (
+	"database/sql"
+	"html/template"
+	"log"
+	"net/http"
+	
+	"manav402/server/DB"
+	"manav402/server/models"
+)
+
+func panicRecovery(res http.ResponseWriter, req *http.Request) {
+	if recover() != nil {
+		res.WriteHeader(http.StatusInternalServerError)
+		res.Header().Set("content-type", "text/json")
+		res.Write([]byte(`{code:500,message:"server error"}`))
+	}
+}
+
+// a handler function for route /register which store profile data in database
+// @params :- a response writer to write data back to client and req which give us header file from client
+func HandleRegsiter(res http.ResponseWriter, req *http.Request) {
+	
+	defer panicRecovery(res,req)
+	var err error
+	// in case some unusual happens the server will send error data
+	
+	// calling parse form to decrypt the form data
+	err = req.ParseForm()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	// storing created data to user profile structure
+	var userData = models.Profile{
+		Fname:  req.FormValue("first_name"),
+		Lname:  sql.NullString{String: req.FormValue("last_name")},
+		Dob:    req.FormValue("dob"),
+		Email:  req.FormValue("email"),
+		Number: req.FormValue("mo_number"),
+	}
+
+	// calling insert function to store data in database
+	err = DB.Insert(userData)
+	if err != nil {
+		log.Panic(err)
+		return
```

Comment:
return error in api.
{
"status":"error",
"message":"error message dynamic"
}